lib/puppet/provider/group/directoryservice.rb

```diff
@@ -9,4 +9,14 @@
   confine :operatingsystem => :darwin
   defaultfor :operatingsystem => :darwin
   has_feature :manages_members
+
+  def members_insync?(current, should)
+    return false unless current
+    if current.respond_to?(:sort) and should.respond_to?(:sort)
```

Comment:
The `should` parameter should always be an array. If the `should` value is nil, e.g. the desired value wasn't specified in the manifest, then puppet will short circuit evaluation earlier in [`safe_insync?`](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/property.rb#L256), since we're not managing that property.

If the `should` value is not an array, then the [default `insync?` logic fail hard](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/property.rb#L291).

Ideally, the `current` parameter would also always be an array. However, it appears that if the group currently has no members, then the provider has a bug and sends in current as `:absent`, instead of an empty array.